extmod/modframebuf.c

```diff
@@ -273,42 +273,59 @@ STATIC void fill_rect(const mp_obj_framebuf_t *fb, int x, int y, int w, int h, u
 STATIC mp_obj_t framebuf_make_new(const mp_obj_type_t *type, size_t n_args, size_t n_kw, const mp_obj_t *args_in) {
     mp_arg_check_num(n_args, n_kw, 4, 5, false);
 
-    mp_obj_framebuf_t *o = mp_obj_malloc(mp_obj_framebuf_t, type);
-    o->buf_obj = args_in[0];
-
-    mp_buffer_info_t bufinfo;
-    mp_get_buffer_raise(args_in[0], &bufinfo, MP_BUFFER_WRITE);
-    o->buf = bufinfo.buf;
+    mp_int_t width = mp_obj_get_int(args_in[1]);
+    mp_int_t height = mp_obj_get_int(args_in[2]);
+    mp_int_t format = mp_obj_get_int(args_in[3]);
+    mp_int_t stride = n_args >= 5 ? mp_obj_get_int(args_in[4]) : width;
 
-    o->width = mp_obj_get_int(args_in[1]);
-    o->height = mp_obj_get_int(args_in[2]);
-    o->format = mp_obj_get_int(args_in[3]);
-    if (n_args >= 5) {
-        o->stride = mp_obj_get_int(args_in[4]);
-    } else {
-        o->stride = o->width;
+    if (width < 1 || height < 1 || width > 0xffff || height > 0xffff || stride > 0xffff || stride < width) {
+        mp_raise_ValueError(MP_ERROR_TEXT("bounds"));
```

Comment:
Did you try making the error message just NULL?

Reply:
-8 bytes (it's not already in the compressed table).